src/clarity-angular/datagrid/datagrid-column.ts

```diff
@@ -87,36 +100,92 @@ export class DatagridColumn extends DatagridFilterRegistrar<DatagridStringFilter
         } else if (value && !this.sorted) {
             this.sort();
         }
-    };
+    }
 
+    /**
+     * @deprecated This will be removed soon, in favor of the sortOrder mechanism
+     */
     @Output("clrDgSortedChange") public sortedChange = new EventEmitter<boolean>();
 
+    // deprecated: to be removed - END
+
+    /**
+     * Indicates how the column is currently sorted
+     */
+    private _sortOrder: SortOrder = SortOrder.Unsorted;
+    public get sortOrder() {
+        return this._sortOrder;
+    }
+
+    @Input("clrDgSortOrder")
+    public set sortOrder(value: SortOrder) {
+        // only if the incoming order is different from the current one
+        if (this._sortOrder === value) {
+            return;
+        }
+
+        switch (value) {
+            // the Unsorted case happens when the current state is either Asc or Desc
+            default: case SortOrder.Unsorted:
+                this._sort.clear();
+                break;
+            case SortOrder.Asc:
+                this.sort(false);
+                break;
+            case SortOrder.Desc:
+                this.sort(true);
+                break;
+        }
+    };
+
+    @Output("clrDgSortOrderChange") public sortOrderChange = new EventEmitter<SortOrder>();
+
     /**
      * Sorts the datagrid based on this column
      */
-    public sort() {
+    public sort(reverse?: boolean) {
         if (!this.sortable) {
             return;
         }
+
+        this._sort.toggle(this.sortBy, reverse);
+
+        // setting the private variable to not retrigger the setter logic
+        this._sortOrder = this._sort.reverse ? SortOrder.Desc : SortOrder.Asc;
+        this.sortOrderChange.emit(this._sortOrder);
```

Comment:
This is more of a stylistic comment than anything and nothing I would go out of my way to correct if you don't want to.

But it appears we are emitting `sortOrderChange` in a few places in this code. My preference is to focus event emission in one location if possible.

I mean, given the subscription on `_sort.change` wouldn't this component fire multiple emissions for one sort change?